### DISCUSSION TITLE
Running multiple instances of wyze-bridge

### DISCUSSION THREAD

**Discussion by carldanley** [mrlt8/docker-wyze-bridge#1385] [Q&A]
At home, I currently have 8 wyze cameras that I connect into a frigate install via wyze-bridge (been happy with it and running a long time now). I recently noticed that the wyze-bridge pod (in my Kubernetes cluster)  is being OOM killed. I had the limit set to 8GB and it seems to need about 16GB to work well (which is ok because the nodes in my cluster have 64GB each). My wyze-bridge pod also leverages hardware acceleration and each node in my cluster can provide a GPU for this functionality.

My initial gut reaction was to just raise the limit of memory that wyze-bridge can consume (which I did) but then I got to thinking... **What if I could deploy an instance of wyze-bridge per camera?** This is a little extra work for me to manage (since I need to now setup `n` number of deployments; 1 per camera) BUT the gains are actually big for me. With this setup, **I'd be able to load balance the CPU/Memory/GPU requirements across 5 nodes in the cluster; thus, reducing the load on 1 single node and allowing Kubernetes to float pods around based on resource allocations, etc (as opposed to forcing 1 big load on any given node).**

My only question is: If I were to run `n` number of instances of wyze-bridge - **do I need to worry about API rate limiting?** I suppose I could always get around this by setting the cameras up on separate accounts but I don't really want to do that if I don't have to.

**Comment by dmkjr** [comment]
rate limited from Wyze?  You will run into an issue with the rate doing it that way.  That is the reason the live feed's and motion detection was removed from the other Wyze Home Assistant integration.

**Comment by carldanley** [comment]
![image](https://github.com/user-attachments/assets/85130336-49af-4476-9a7f-b94f57ef8ae6)

-- [Source](https://support.wyze.com/hc/en-us/articles/16129834216731-Creating-an-API-Key)

If I'm using local cameras and the initial access token lasts 2 days and refresh tokens last 30 days - what other calls are made? That is what I'm trying to understand.

I obviously brought up the rate limiting bit and am aware I'd probably run into it but was hoping for specifics without having to spend time looking into every API call in the code.

EDIT: I don't use the Wyze UI / Object Detection (from Wyze) / or anything else in that way (using camera commands, etc). Wyze is deployed and literally used for local rtsp streaming -- frigate takes care of the rest.
